beetsplug/lastgenre/__init__.py

```diff
@@ -111,39 +111,41 @@ def setup(self):
             self.import_stages = [self.imported]
 
         self._genre_cache = {}
+        self.whitelist = self._load_whitelist()
+        self.c14n_branches, self.canonicalize = self._load_c14n_tree()
 
-        # Read the whitelist file if enabled.
-        self.whitelist = set()
+    def _load_whitelist(self) -> set[str]:
+        whitelist = set()
         wl_filename = self.config["whitelist"].get()
         if wl_filename in (True, ""):  # Indicates the default whitelist.
             wl_filename = WHITELIST
         if wl_filename:
             wl_filename = normpath(wl_filename)
             with open(wl_filename, "rb") as f:
-                for line in f:
-                    line = line.decode("utf-8").strip().lower()
+                for raw_line in f:
+                    line = raw_line.decode("utf-8").strip().lower()
                     if line and not line.startswith("#"):
-                        self.whitelist.add(line)
+                        whitelist.add(line)
+        return whitelist
 
-        # Read the genres tree for canonicalization if enabled.
-        self.c14n_branches = []
+    def _load_c14n_tree(self) -> tuple[list[list[str]], bool]:
+        c14n_branches: list[list[str]] = []
```

Comment:
Also here, please double check if that is the correct way to do it. I fixed this one:

```
beetsplug/lastgenre/__init__.py:132: error: Need type annotation for "c14n_branches" (hint: "c14n_branches: list[<type>] = ...")  [var-annotated]
```

Reply:
also here decision that it's ok. resolved.